go-app.js

```diff
@@ -45,7 +45,18 @@ go.utils = {
             sum += +c;
         }
         sum = 10 - ('' + sum).charAt(1);
-        return ('' + sum).slice(-1) == check;
+        if( ('' + sum).slice(-1) == check){
+            //the id_no is valid. NOw check that valid will accept the date
+            var month,day,year;
+            year = parseInt( id.substr(0,2));
+            month =parseInt( id.substr(2,2)); 
```

Comment:
please watch your spacing and be consistent. `var month = parseInt(id.substr(2,2));` is better.